samples/nextjs/src/lib/page-props-factory.ts

```diff
@@ -0,0 +1,149 @@
+import { AxiosError } from 'axios';
+import { ParsedUrlQuery } from 'querystring';
+import { GetServerSidePropsContext, GetStaticPropsContext, NextPageContext } from 'next';
+import {
+  ComponentPropsService,
+  EditingRequest,
+  LayoutService,
+  LayoutServiceData,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { SitecorePageProps } from 'lib/page-props';
+import { componentModule } from 'temp/componentFactory';
+import { layoutService } from 'lib/layout-service';
+import { DictionaryService, dictionaryService } from 'lib/dictionary-service';
+import { config as packageConfig } from '../../package.json';
+
+/**
+ * Extract normalized Sitecore item path from query
+ */
+const extractPath = function (params: ParsedUrlQuery | undefined): string {
+  if (params === undefined) {
+    return '/';
+  }
+  let path = Array.isArray(params.path) ? params.path.join('/') : params.path ?? '/';
+
+  // Ensure leading '/'
+  if (!path.startsWith('/')) {
+    path = '/' + path;
+  }
+
+  return path;
+};
+
+/**
+ * Determine if Next.js context is GetServerSidePropsContext
+ * @param context {GetStaticPropsContext|GetServerSidePropsContext}
+ */
+const isGetServerSidePropsContext = function (
+  context: GetServerSidePropsContext | GetStaticPropsContext
+): context is GetServerSidePropsContext {
+  return 'req' in context;
+};
+
+export class SitecorePagePropsFactory {
+  private get componentPropsService(): ComponentPropsService {
```

Comment:
@ambrauer Do we need to create new instance of `ComponentPropsService` every time?

Reply:
@sc-illiakovalenko Moved to private var / constructor init.